cmd/ingress/main.go

```diff
@@ -134,17 +147,40 @@ func (env *envConfig) send(event *cloudevents.Event) (int, error) {
 	for key, val := range event.Extensions() {
 		headers[key] = val
 	}
-	if err := env.channel.Publish(
-		env.ExchangeName,
-		"",    // routing key
-		false, // mandatory
-		false, // immediate
-		amqp.Publishing{
-			Headers:     headers,
-			ContentType: "application/json",
-			Body:        bytes,
-		}); err != nil {
+
+	ch, err := env.sendMessage(headers, bytes)
+
+	if err != nil {
 		return http.StatusInternalServerError, fmt.Errorf("failed to publish message")
 	}
-	return http.StatusAccepted, nil
+
+	confirmation := <-ch
+
+	if confirmation.Ack {
+		return http.StatusAccepted, nil
+	} else {
+		return http.StatusServiceUnavailable, errors.New("message was not confirmed")
+	}
+}
+
+func (env *envConfig) sendMessage(headers amqp.Table, bytes []byte) (chan amqp.Confirmation, error) {
+	crc := atomic.AddUint64(&env.rc, 1)
```

Comment:
looks like this means the first message will bump counter to 1 and hence env.channels[0] would never be used? if so, you might change your slice indexing in line 175 accordingly